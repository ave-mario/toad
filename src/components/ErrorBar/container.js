import { connect } from 'react-redux';
import ErrorBar from './view';

const mapStateToProps = state => ({
  message: state.error.message || state.error.success,
  isError: Boolean(state.error.message)
});

export default connect(mapStateToProps)(ErrorBar);
