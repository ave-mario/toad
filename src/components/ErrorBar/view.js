import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Close from '@material-ui/icons/Close';
import { ServerError, ServerErrorText } from 'elements';
import services from 'services';

class ErrorBar extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      isShow: true,
      message: props.message
    };
    this.handleClick = this.handleClick.bind(this);
  }

  static getDerivedStateFromProps(nextProps, prevState) {
    if (prevState.message === nextProps.message) return { isShow: true };
    return null;
  }

  changeError() {
    const { message, isError } = this.props;
    if (isError) return services.errorHandlerService(message);
    return message;
  }

  handleClick() {
    this.setState(prevState => ({ isShow: !prevState.isShow }));
  }

  render() {
    const { isShow } = this.state;
    const { isError } = this.props;
    const message = this.changeError();
    return (
      <>
        {message && isShow && (
          <ServerError fixed backgroundColor={isError ? '#ed1e3a' : '#27ae60'}>
            <ServerErrorText>{message}</ServerErrorText>
            <Close
              style={{ cursor: 'pointer', color: 'white', fontSize: '30' }}
              onClick={this.handleClick}
            />
          </ServerError>
        )}
      </>
    );
  }
}

ErrorBar.propTypes = {
  message: PropTypes.oneOfType([
    PropTypes.objectOf(PropTypes.oneOfType([PropTypes.string, PropTypes.bool])),
    PropTypes.bool,
    PropTypes.string
  ]),
  isError: PropTypes.bool.isRequired
};

ErrorBar.defaultProps = {
  message: ''
};
export default ErrorBar;
