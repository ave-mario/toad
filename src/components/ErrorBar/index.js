import ErrorBar from './container';

export default ErrorBar;
