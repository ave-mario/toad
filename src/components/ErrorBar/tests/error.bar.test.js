import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import ErrorBar from '../container';
import 'jest-styled-components';

test('ErrorBar should render correctly', () => {
  const mockStore = configureStore();
  const store = mockStore({
    auth: {
      isFailed: false,
      user: {}
    },
    error: { message: '' }
  });

  const wrapper = mount(
    <Provider store={store}>
      <ErrorBar />
    </Provider>
  );
  expect(wrapper).toMatchSnapshot();
});
