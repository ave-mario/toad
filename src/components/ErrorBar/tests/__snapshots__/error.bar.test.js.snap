// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBar should render correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(ErrorBar)>
    <ErrorBar
      dispatch={[Function]}
      isError={false}
      message=""
    />
  </Connect(ErrorBar)>
</Provider>
`;
